server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -147,6 +148,8 @@
 @LuceneTestCase.SuppressFileSystems("ExtrasFS")
 public class TranslogTests extends ESTestCase {
```

Comment:
I think we need to test that reading data in the old format works? I suppose we have release upgrade tests that do so superficially (not sure if we are guaranteed they'll read all types from translog) but it would be nice to have an explicit test for it.